src/ClassifierHelpers.ts

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable security/detect-object-injection */
+// Disabling this since the standard object we will pass will have no prototype chain
+// thus, it will return undefined for properties not present in the object itself
+
+/**
+ * Changes the last element of array to `*`
+ * @param path The path in form of an array
+ * @returns The path in form of a string with `*` at the end
+ *
+ * @example
+ * changeLastElementToPlaceholder(['', 'servers', 'production']); // '/servers/*'
+ */
+export function changeLastElementToPlaceholder(path: string[]): string {
+  path.pop();
+  path.push('*');
+  return path.join('/');
+}
+
+/**
+ * Generates the correct path to get the classification data from standard
+ * @param standard The standard object
+ * @param path The path to find
+ * @returns The path found
+ */
+export function generateClassifierPath(
+  standard: any,
+  path: string
```

Comment:
path must be valid json pointer, otherwise you will have a problem when the path points to the channel where channel name contains `/`.

JSON Pointer will make `/` look like `~1`. Example: where channel name is `smartylighting/streetlights/1/0/action/{streetlightId}/turn/off` ([link](https://github.com/asyncapi/parser-js/blob/master/test/wrong/invalid-asyncapi-with-ref-with-line-break.yaml#L64)) -> `/channels/smartylighting~1streetlights~11~10~1action~1{streetlightId}~1turn~1off/parameters/streetlightId/$ref`

There are some helpers in the parser already that might be useful https://github.com/asyncapi/parser-js/blob/master/lib/utils.js

json pointer transformation, tilde/untilde

Reply:
Yeah, the diffing function we are using [here](https://github.com/asyncapi/diff/pull/18) will escape these path strings. So it's safe to use it here.

Reply:
ah perfect, it outputs json pointer ❤️ 
shouldn't it be explicit here in the jsdoc?

Reply:
oh yeah, will add that in the jsdoc. Thanks :)